src/__test__/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapLegendVisibilitySettings.test.jsx

```diff
@@ -0,0 +1,166 @@
+            key: 'louvain-0',
+          },
+          {
+            key: 'louvain-1',
+          },
+        ],
+      },
+      {
+        key: 'sample',
+        children: [
+          {
+            key: 'control',
+          },
+          {
+            key: 'treated',
+          },
+        ],
+      },
+    ],
+    properties: {
+      louvain: {
+        type: 'cellSets',
+        name: 'louvain clusters',
+      },
+      'louvain-0': {
+        name: 'louvain 0',
+        cellIds: [5, 6, 7],
+      },
+      'louvain-1': {
+        name: 'louvain 1',
+        cellIds: [1, 2, 3],
+      },
+      sample: {
+        type: 'metadataCategorical',
+        name: 'Sample',
+      },
+      control: {
+        name: 'control',
+        cellIds: [5, 6, 7],
+      },
+      treated: {
+        name: 'treated',
+        cellIds: [1, 2, 3],
+      },
+    },
+  },
+  cellInfo: {},
+  componentConfig: {
+    interactiveHeatmap: {
+      config: {
+        groupedTrack: 'sample',
+        selectedTracks: ['louvain'],
+        legendIsVisible: true,
+      },
+    },
+  },
+};
+
+const mockStore = configureMockStore([thunk]);
+configure({ adapter: new Adapter() });
+
+const eventStub = {
+  stopPropagation: () => { },
+};
+
+describe('HeatmapLegendVisibilitySettings', () => {
+  let component, store, buttons;
+
+  afterEach(() => {
+    component.unmount();
+  });
+
+  beforeEach(() => {
+    store = mockStore({
+      ...initialState,
+    });
+
+    component = mount(
+      <Provider store={store}>
+        {/* ask if experimentId='123' width={200} height={200} are necessary in HeatmapGroupBySettings */}
+        <HeatmapLegendVisibilitySettings experimentId='123' width={200} height={200} />
+      </Provider>,
+    );
+  });
+
+  it('renders correctly', () => {
+    const buttons = component.find('Radio');
+
+    // Should be rendered.
+    expect(component.find('HeatmapLegendVisibilitySettings').length).toEqual(1);
+
+    // With two buttons.
+    expect(buttons.length).toEqual(2);
+
+    // With the right keys.
+    expect(buttons.at(0).props().value).toEqual(true);
+    expect(buttons.at(1).props().value).toEqual(false);
+
+    // The selected one should be `Show`.
+    expect(component.find(Radio.Group).props().value).toEqual(true);
+  });
+
+  it('responds correctly to clicking to hide', () => {
+    // Click hide button
+    act(() => {
+      component.find('ForwardRef').props().onChange({ "target": { "value": false } });
+    });
+
+    let storeActions = store.getActions();
+
+    // Only one action was dispatched
+    expect(storeActions.length).toEqual(1);
+
+    let action = storeActions[0];
+    let configChange = action.payload.configChange;
+
+    // Of the correct type
+    expect(action.type).toBe(UPDATE_CONFIG);
+
+    // With the correct property
+    expect(configChange).toHaveProperty("legendIsVisible", false);
+
+    // QUESTION: Should I check that the object doesn't have any other property? (example: check size is 1)
```

Comment:
if you feel that this is important for the correctness of this test, then you should. I personally don't see any reason why not checking

Reply:
In general, it's a good idea to check the minimum set of features needed for this system to work. This is so that you don't have to re-write tests if extraneous details get updated. This can be especially annoying with Redux.

My suggestion is that if the legendIsVisible property being true is necessary and sufficient in hiding and showing the legend, you should not test any other property of the object, at least not explicitly.

What you should do is add a condition on a matching snapshot -- see here -- this will explicitly save the state of the object into a file and when it changes, it will warn you and fail the condition, but you can easily update the snapshot with a command if the change was intentional.